File: ai/src/main/scala/org/pingel/gestalt/core/Name.java
package org.pingel.gestalt.core;

public class Name implements Comparable
{
    static int counter = 0;

    public String base;

    public Name()
    {
		base = "s" + Integer.toString(counter);
		counter++;
    }

    public Name(String n)
    {
		base = n;
    }
    
    public String toString() 
    {
		return base;
    }
    
    public int compareTo(Object obj)
    {
		Name other = (Name) obj;
		return base.compareTo(other.base);
    }

    public boolean equals(Object obj)
    {
		return base.equals(((Name)obj).base);
    }

}

File: ai/src/main/scala/org/pingel/gestalt/core/Name.scala
package org.pingel.gestalt.core

object Name {
  var count = 0
  def nextName() = {
    val result = "s" + count
    count += 1
    result
  }
}

case class Name(var base: String=null) extends Comparable[Name]
{
	if( base == null ) {
	  base = Name.nextName()
	}

    def toString() = base

    def compareTo(obj: Object) = {
		val other = obj.asInstanceOf[Name]
		base.compareTo(other.base)
    }

    // runtime error in asInstanceOf:
    def equals(obj: Object) = base.equals(obj.asInstanceOf[Name].base)

}
